src/OpenTelemetrySDK/OpenTelemetryController.cs

```diff
@@ -0,0 +1,152 @@
+
+            // If we get at least one result, we know the OpenTelemetry SDK was imported
+            var numCandidates = matchingModules.Count();
+            var isModuleInCurrentSession = numCandidates > 0;
+
+            if (isModuleInCurrentSession)
+            {
+                var candidatesInfo = matchingModules.Select(module => string.Format(
+                    PowerShellWorkerStrings.FoundOpenTelemetrySdkInSession, module.Name, module.Version, module.Path));
+                var otelSDKModuleInfo = string.Join('\n', candidatesInfo);
+
+                if (numCandidates > 1)
+                {
+                    // If there's more than 1 result, there may be runtime conflicts
+                    // warn user of potential conflicts
+                    _logger.Log(isUserOnlyLog: false, LogLevel.Warning, String.Format(
+                        PowerShellWorkerStrings.MultipleExternalSDKsInSession,
+                        numCandidates, Utils.OpenTelemetrySdkName, otelSDKModuleInfo));
+                }
+                else
+                {
+                    // a single SDK is in session. Report its metadata
+                    _logger.Log(isUserOnlyLog: false, LogLevel.Trace, otelSDKModuleInfo);
+                }
+            }
+
+            _isOpenTelemetryModuleLoaded = isModuleInCurrentSession;
+
+
+            return _isOpenTelemetryModuleLoaded.Value;
+        }
+
+        public static void ResetOpenTelemetryModuleStatus()
+        {
+            _isOpenTelemetryCapable = null;
+            _isOpenTelemetryModuleLoaded = null;
+            _isOpenTelmetryEnvironmentEnabled = null;
+        }
+
+        public void AddStartOpenTelemetryInvocationCommand(OpenTelemetryInvocationContext otelContext)
+        {
+            if (!isOpenTelemetryCapable() || !otelContext.isValid())
+            {
+                return;
+            }
+
+            _pwsh.AddCommand(StartOpenTelemetryInvocationCmdlet)
+                .AddParameter("InvocationId", otelContext.InvocationId)
+                .AddParameter("TraceParent", otelContext.TraceParent)
+                .AddParameter("TraceState", otelContext.TraceState);
+        }
+
+        public void StopOpenTelemetryInvocation(OpenTelemetryInvocationContext otelContext)
+        {
+            if (!isOpenTelemetryCapable())
+            {
+                return;
+            }
+
+            _pwsh.AddCommand(StopOpenTelemetryInvocationCmdlet)
+                .AddParameter("InvocationId", otelContext.InvocationId)
+                .InvokeAndClearCommands();
+        }
+
+        internal void StartFunctionsLoggingListener()
+        {
+            if (!isOpenTelemetryCapable())
+            {
+                return;
+            }
+
+            var eventHandlers = _pwsh.AddCommand(GetFunctionsLogHandlerCmdlet)
+                .InvokeAndClearCommands<Action<string, string, Exception>>();
+
+            foreach (var eventHandler in eventHandlers)
```

Comment:
Do you really anticipate more than one item in `eventHandlers`? If not, perhaps you don't need a loop and a collection in `outputLogHandler`? At least not _yet_?

Reply:
True - this was more future-proofing than anything. I can minimize this for now

Reply:
Minimized to single-handler set to first response from cmdlet